backend/src/modules/tournament-tree/core/domain/TournamentTree.ts

```diff
@@ -72,8 +74,13 @@ export class TournamentTree {
     return [...this.tournamentTreeArray];
   }
 
-  public getMatchesQueueReadyToBegin(): FightingTeamsGroup[] {
-    return this.tournamentTreeArray.filter(({ firstTeam, secondTeam }) => firstTeam && secondTeam);
+  public getMatchesQueueReadyToStart(): MatchReadyToStart[] {
+    return this.tournamentTreeArray.filter(({ firstTeam, secondTeam }) => firstTeam && secondTeam).map((match) => {
+        return new MatchReadyToStart(
+            { firstTeam: match.firstTeam, secondTeam: match.secondTeam, matchNumber: match.matchNumberInSequence }
```

Comment:
Ekipo niech mnie ktoś oświeci z tym mapowaniem typów! Albo jestem bardzo zmęczony albo nie rozumiem, skoro filter znajduje mi tylko takie mecze, które mają zagwarantowane pola firstTeam i secondTeam (!= undefined) to czemu do piekła TS nie bierze tego jako pewnik do późniejszego mapowania typów i ciągle mam 'Object is possibly undefined' przy polach MatchReadyToStart?

Reply:
Dobra, poczytałem i już wiem dlaczego tak się dzieje, ale dalej nie wiem jak to zgrabnie ominąć nie używając ! lub 'as' :P HELP

Reply:
Teraz chyba nie dasz rady tego ominąć :) Ale to jest spoko, bo to szczegół implementacyjny. Masz to w środku klasy - metody. A na zewnątrz wychodzi ładny typ i można go używać.